### PR TITLE
more config v6 tests

### PR DESCRIPTION
**Pull request by kp-cat**
### Describe the purpose of your pull request

Adds a few additional test cases for testing config v6 features.
Also, fixes a few edge cases where user attribute values was not converted to their canonical string representations.

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [x] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

**Review by adams85 — CHANGES_REQUESTED** [review]
https://github.com/configcat/python-sdk/blob/bbd5128e38c4cb4a23349682ec5f4e10325835bf/configcatclient/user.py#L89

Let's use the `ensure_ascii=False` option here as well.

**Review by adams85 — CHANGES_REQUESTED** [review]
Thanks for implementing the additional tests!

Only a few minor inconsistencies are left to polish off.
* Please update the message of log event 1103 as shown here: https://github.com/configcat/common-js/pull/100/commits/50aeed44c1fde5ec4155bad5e7f9b84830a5b595
* Please implement this consistency improvement change: https://github.com/configcat/elixir-sdk/commit/c22c00c91066b4db332dbdf88e42d44cc4d9dd4a (discussion: https://github.com/configcat/elixir-sdk/pull/133#discussion_r14501412)
* It would be nice to address this as well: https://github.com/configcat/elixir-sdk/pull/133#discussion_r1471830473 In Elixir, it was a pretty easy change: https://github.com/configcat/elixir-sdk/commit/43d9d89a99b44875fe97f3acc801ecbcd8fc7c90 If it's not too hard to do this in Python, please include this change as well.

**Review by adams85 — APPROVED** [review]

File: configcatclient/config.py [hunk 1]
```diff
@@ -61,6 +63,24 @@
 UNSUPPORTED_VALUE = 'unsupported_value'
 
 
+def is_type_missmatch(value, py_type):
+    is_float_int_missmatch = \
+        (type(value) is float and py_type is int) or \
+        (type(value) is int and py_type is float)
+
+    # On Python 2.7, ignore the type missmatch is between str and unicode.
```
Comment (kp-cat) [inline configcatclient/config.py:71]:
```suggestion
    # On Python 2.7, ignore the type mismatch between str and unicode.
```
